packages/clerk-js/src/ui/elements/Switch.tsx

```diff
@@ -0,0 +1,118 @@
+import React, { forwardRef, useState } from 'react';
+
+import type { LocalizationKey } from '../customizables';
+import { descriptors, Flex, Text } from '../customizables';
+
+interface SwitchProps {
+  checked?: boolean;
+  defaultChecked?: boolean;
+  onChange?: (checked: boolean) => void;
+  disabled?: boolean;
+  'aria-label'?: string;
+  'aria-labelledby'?: string;
+  title?: string | LocalizationKey;
+}
+
+export const Switch = forwardRef<HTMLDivElement, SwitchProps>(
+  (
+    {
+      checked: controlledChecked,
+      defaultChecked,
+      onChange,
+      disabled = false,
+      'aria-label': ariaLabel,
+      'aria-labelledby': ariaLabelledby,
+      title,
+    },
+    ref,
+  ) => {
+    const [internalChecked, setInternalChecked] = useState(!!defaultChecked);
+    const isControlled = controlledChecked !== undefined;
+    const checked = isControlled ? controlledChecked : internalChecked;
+
+    const handleToggle = (_: React.MouseEvent | React.KeyboardEvent) => {
+      if (disabled) return;
+      if (!isControlled) {
+        setInternalChecked(!checked);
+      }
+      onChange?.(!checked);
+    };
+
+    const handleKeyDown = (e: React.KeyboardEvent) => {
+      if (e.key === ' ' || e.key === 'Enter') {
+        e.preventDefault();
+        handleToggle(e);
+      }
+    };
+
+    return (
+      <Flex
+        direction='row'
+        align='center'
+        as='div'
+        ref={ref}
+        tabIndex={disabled ? -1 : 0}
+        role='switch'
+        aria-checked={checked}
+        aria-label={ariaLabel}
+        aria-labelledby={ariaLabelledby}
```

Comment:
I think we don't need one or the other here. `ariaLabelledby` would be applied to the text right?